object_detectors/detectron2_lmi/model.py

```diff
@@ -0,0 +1,518 @@
+        return predictions
+    
+    def annotate_image(self, result, image, color_map=None):
+        """
+        Annotates an image with bounding boxes, class labels, and masks.
+
+        Args:
+            result (dict): A dictionary containing detection results with keys:
+                - "classes" (list): List of class labels for detected objects.
+                - "boxes" (list): List of bounding boxes for detected objects.
+                - "scores" (list): List of confidence scores for detected objects.
+                - "masks" (list, optional): List of masks for detected objects.
+            image (numpy.ndarray): The image to annotate.
+            color_map (dict, optional): A dictionary mapping class labels to colors.
+
+        Returns:
+            numpy.ndarray: The annotated image.
+        """
+        for i in range(len(result["classes"])):
+            plot_one_box(
+                result["boxes"][i],
+                image,
+                label=f"{result['classes'][i]}:{result['scores'][i]:.2f}",
+                mask=result["masks"][i] if len(result["masks"]) > 0 else None,
+                color=color_map,
+            )
+        return image
+
+class Detectron2Torchscript(ModelBase):
+    logger = logging.getLogger(__name__)
+   
+    def __init__(self, model_path, class_map, device='cuda', **kwargs):
+        try:
+            self.model = torch.jit.load(model_path)
+        except Exception as e:
+            self.logger.exception(f"Failed to load model: {e}")
+        
+        # move the model to gpu
+        self.model.to(device)
+        
+        self.class_map = {
+            int(k): str(v) for k, v in class_map.items()
+        }
+        self.batch_size = kwargs.get('batch_size', 1)
+        self.class_map_func = np.vectorize(lambda c: self.class_map.get(int(c), str(c)))
+    
+    def warmup(self, **kwargs):
+        """
+        Perform a warmup operation for the model.
+
+        This method generates a random input tensor with the specified image size and 
+        passes it through the model to perform a warmup. This can be useful to initialize 
+        model parameters and optimize performance before actual inference.
+
+        Args:
+            **kwargs: Arbitrary keyword arguments.
+                img_size (tuple, optional): A tuple specifying the height and width of the image.
+                                            If not provided, an error is logged.
+
+        Raises:
+            ValueError: If 'img_size' is not provided in kwargs.
+        """
+        image_size = kwargs.get('img_size', [1024,1024])
+        image_h, image_w = image_size[0], image_size[1]
+        images = [np.random.rand(image_h, image_w, 3).astype(np.float32) for _ in range(self.batch_size)]
+        input = [
+            dict(image=torch.from_numpy(image.astype(np.float32)).permute(2, 0, 1).to(dtype=torch.float32).cuda()) for image in images
```

Comment:
if the user pass device='cpu' in the class's constructor for a demo system, this would generate errors. Instead of using cuda(), it would be better to send to a self.device attribute.

Reply:
removed using .cuda() using .to(device)